shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java

```diff
@@ -2049,28 +2051,122 @@ public void ime_windowInsetsSync() {
     WindowInsets.Builder builder = new WindowInsets.Builder();
     WindowInsets noneInsets = builder.build();
 
-    // imeInsets0, 1, and 2 contain unique IME bottom insets, and are used
-    // to distinguish which insets were sent at each stage.
-    builder.setInsets(WindowInsets.Type.ime(), Insets.of(0, 0, 0, 100));
-    builder.setInsets(WindowInsets.Type.navigationBars(), Insets.of(10, 10, 10, 40));
+    builder.setInsets(WindowInsets.Type.ime(), Insets.of(0, 0, 0, 25));
+    builder.setInsets(WindowInsets.Type.navigationBars(), Insets.of(0, 0, 0, 40));
     WindowInsets imeInsets0 = builder.build();
 
-    builder.setInsets(WindowInsets.Type.ime(), Insets.of(0, 0, 0, 30));
-    builder.setInsets(WindowInsets.Type.navigationBars(), Insets.of(10, 10, 10, 40));
+    builder.setInsets(WindowInsets.Type.ime(), Insets.of(0, 0, 0, 50));
+    builder.setInsets(WindowInsets.Type.navigationBars(), Insets.of(0, 0, 0, 40));
     WindowInsets imeInsets1 = builder.build();
 
+    builder.setInsets(WindowInsets.Type.ime(), Insets.of(0, 0, 0, 100));
+    builder.setInsets(WindowInsets.Type.navigationBars(), Insets.of(0, 0, 0, 0));
+    WindowInsets deferredInsets = builder.build();
+
     builder.setInsets(WindowInsets.Type.ime(), Insets.of(0, 0, 0, 50));
-    builder.setInsets(WindowInsets.Type.navigationBars(), Insets.of(10, 10, 10, 40));
-    WindowInsets imeInsets2 = builder.build();
+    builder.setInsets(WindowInsets.Type.navigationBars(), Insets.of(0, 0, 0, 40));
+    WindowInsets altDeferredInsets = builder.build();
+
+    ArgumentCaptor<FlutterRenderer.ViewportMetrics> viewportMetricsCaptor =
+        ArgumentCaptor.forClass(FlutterRenderer.ViewportMetrics.class);
+
+    // Set the initial insets
+    imeSyncCallback.getInsetsListener().onApplyWindowInsets(testView, noneInsets);
+
+    verify(flutterRenderer, atLeast(1)).setViewportMetrics(viewportMetricsCaptor.capture());
+    assertEquals(0, viewportMetricsCaptor.getValue().viewPaddingBottom);
+    assertEquals(0, viewportMetricsCaptor.getValue().viewPaddingTop);
+    assertEquals(0, viewportMetricsCaptor.getValue().viewInsetBottom);
+    assertEquals(0, viewportMetricsCaptor.getValue().viewInsetTop);
+
+    // Set the deferred insets and make a second call after the animation start which should be
+    // ignored
+    imeSyncCallback.getAnimationCallback().onPrepare(animation);
+    imeSyncCallback.getInsetsListener().onApplyWindowInsets(testView, deferredInsets);
+    imeSyncCallback.getAnimationCallback().onStart(animation, null);
+    imeSyncCallback.getInsetsListener().onApplyWindowInsets(testView, altDeferredInsets);
+
+    verify(flutterRenderer, atLeast(1)).setViewportMetrics(viewportMetricsCaptor.capture());
+    assertEquals(0, viewportMetricsCaptor.getValue().viewPaddingBottom);
+    assertEquals(0, viewportMetricsCaptor.getValue().viewPaddingTop);
+    assertEquals(0, viewportMetricsCaptor.getValue().viewInsetBottom);
```

Comment:
Can you add a comment here explaining that the 0 here is the "real" test. The other values are echoing the many 0's set above whereas I think it is the viewInsetBottom that is the valuable test. 

This (and below) can also be more readable if you used did something like this 
```
assertEquals(deferredInsets.bottom, viewportMetricsCaptor.getValue().viewPaddingBottom);   assertEquals(deferredInsets.top, viewportMetricsCaptor.getValue().viewPaddingTop);
// Ensure that ime insets are not shown when nav bars are rendered on top of the view. 
assertEquals(0, viewportMetricsCaptor.getValue().viewInsetBottom);
assertEquals(deferredInsets.bottom, viewportMetricsCaptor.getValue().viewInsetTop);
```

If I am misreading this code and other constants should be included please correct my example.
There are a lot of zero values and it is a bit difficult to track what each of them means.

Reply:
I've made some changes to the tests which will hopefully improve the clarity and remove the unnecessary checks against other viewport metrics, let me know if there are any additional changes you'd like to see here.